ttnn/cpp/ttnn/operations/pool/generic/generic_pools.cpp

```diff
@@ -2,28 +2,47 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#include "max_pool2d.hpp"
+#include "generic_pools.hpp"
 
 #include "impl/buffers/buffer_constants.hpp"
 #include "ttnn/operations/conv/conv2d/conv2d.hpp"
 #include "ttnn/operations/sliding_window/sliding_window.hpp"
+#include "tt_metal/common/bfloat16.hpp"
 #include "tt_metal/common/math.hpp"
 #include "ttnn/common/constants.hpp"
 
+#include <limits>
+
 namespace ttnn {
 namespace operations::pool {
 
-Tensor MaxPool2DOp::invoke(uint8_t queue_id,
-                           const Tensor& input_tensor,
-                           uint32_t batch_size,
-                           uint32_t input_h, uint32_t input_w,
-                           uint32_t channels,
-                           std::array<uint32_t, 2> kernel_size,
-                           std::array<uint32_t, 2> stride,
-                           std::array<uint32_t, 2> padding,
-                           std::array<uint32_t, 2> dilation,
-                           const std::optional<const MemoryConfig>& memory_config,
-                           const std::optional<const TensorMemoryLayout> applied_shard_scheme) {
+namespace {
+
+// Return a single bf16 init value for the pool type in u32 (packed in the least 16 bits)
+uint32_t get_bf16_pool_init_value(Pool2DType pool_type) {
+    float value;
+    switch (pool_type) {
+        case Pool2DType::MAX_POOL2D: value = -std::numeric_limits<float>::infinity(); break;
```

Comment:
I remember seeing that inf and none values are a bit special and might not directly map to std::numeric_limits.
OP devs know better. Just highlighting this. Will learn from the response to this comment.

Reply:
See the old value that was used `uint32_t neg_inf_pad_val = 0xf7ff;`

Reply:
Yes, we need to convert float32 to bfloat16 for using in kernel. This is done below when we return the value

```cpp
return bfloat16(value).to_packed();
```

The converted bfloat16 is also packed in u32 in order to pass in kernel compile arguments. In kernel we convert (truncate) u32 back to u16 to get the bfloat16. This behaivor is documented in the comment above of this function